test/libsolidity/ABIDecoderTests.cpp

```diff
@@ -463,121 +352,6 @@ BOOST_AUTO_TEST_CASE(validation_function_type)
-	string invalidFun{"01234567890123456789abcdX"};
-	NEW_ENCODER(
-		compileAndRun(sourceCode);
-		ABI_CHECK(callContractFunction("f((function))", validFun), encodeArgs(1));
-		// Error because we copy to memory
-		ABI_CHECK(callContractFunction("f((function))", invalidFun), encodeArgs());
-		ABI_CHECK(callContractFunction("g((function))", validFun), encodeArgs(2));
-		// No error because x is not accessed.
-		ABI_CHECK(callContractFunction("g((function))", invalidFun), encodeArgs(2));
-		ABI_CHECK(callContractFunction("h((function))", validFun), encodeArgs(3));
-		// Error on access.
-		ABI_CHECK(callContractFunction("h((function))", invalidFun), encodeArgs());
-	)
-}
-
-BOOST_AUTO_TEST_CASE(storage_ptr)
-{
-	string sourceCode = R"(
-		library L {
-			struct S { uint x; uint y; }
-			function f(uint[] storage r, S storage s) public returns (uint, uint, uint, uint) {
-				r[2] = 8;
-				s.x = 7;
-				return (r[0], r[1], s.x, s.y);
-			}
-		}
-		contract C {
-			uint8 x = 3;
-			L.S s;
-			uint[] r;
-			function f() public returns (uint, uint, uint, uint, uint, uint) {
-				r = new uint[](6);
-				r[0] = 1;
-				r[1] = 2;
-				r[2] = 3;
-				s.x = 11;
-				s.y = 12;
-				(uint a, uint b, uint c, uint d) = L.f(r, s);
-				return (r[2], s.x, a, b, c, d);
-			}
-		}
-	)";
-	BOTH_ENCODERS(
-		compileAndRun(sourceCode, 0, "L");
-		compileAndRun(sourceCode, 0, "C", bytes(), map<string, Address>{{"L", m_contractAddress}});
-		ABI_CHECK(callContractFunction("f()"), encodeArgs(8, 7, 1, 2, 7, 12));
-	)
-}
-
-BOOST_AUTO_TEST_CASE(struct_simple)
-{
-	string sourceCode = R"(
-		contract C {
-			struct S { uint a; uint8 b; uint8 c; bytes2 d; }
-			function f(S memory s) public pure returns (uint a, uint b, uint c, uint d) {
-				a = s.a;
-				b = s.b;
-				c = s.c;
-				d = uint16(s.d);
-			}
-		}
-	)";
-	NEW_ENCODER(
-		compileAndRun(sourceCode, 0, "C");
-		ABI_CHECK(callContractFunction("f((uint256,uint8,uint8,bytes2))", 1, 2, 3, "ab"), encodeArgs(1, 2, 3, 'a' * 0x100 + 'b'));
-	)
-}
-
-BOOST_AUTO_TEST_CASE(struct_validation)
-{
-	string sourceCode = R"(
-		contract C {
-			struct S { int16 a; uint8 b; bytes2 c; }
-			function f(S memory s) public pure returns (uint a, uint b, uint c) {
-				assembly {
-					a := mload(s)
-					b := mload(add(s, 0x20))
-					c := mload(add(s, 0x40))
-				}
-			}
-		}
-	)";
-	u256 largeNeg("0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff01");
-	NEW_ENCODER(
-		compileAndRun(sourceCode, 0, "C");
-		ABI_CHECK(
-			callContractFunction("f((int16,uint8,bytes2))", largeNeg, 0xff, "ab"),
-			encodeArgs(largeNeg, 0xff, "ab")
-		);
-		ABI_CHECK(
-			callContractFunction("f((int16,uint8,bytes2))", 0xff010, 0xff, "ab"),
-			encodeArgs()
-		);
-		ABI_CHECK(
-			callContractFunction("f((int16,uint8,bytes2))", largeNeg, 0xff0002, "ab"),
-			encodeArgs()
-		);
-		ABI_CHECK(
-			callContractFunction("f((int16,uint8,bytes2))", largeNeg, 0xff, "abcd"),
-			encodeArgs()
-		);
-	)
-}
-
 BOOST_AUTO_TEST_CASE(struct_short)
```

Comment:
Wasn't this moved out, too?

Reply:
I couldn't create exact same encoding via isoltest, and left it here for now.
As there are many more tests that can be moved, in some of next PRs we can move it if possible